common/lib/xmodule/xmodule/modulestore/inheritance.py

```diff
@@ -172,6 +172,14 @@ class InheritanceMixin(XBlockMixin):
         default=True,
         scope=Scope.settings
     )
+    video_auto_advance = Boolean(
```

Comment:
NVM, @sstack22 and I discussed, may be this gets out of this PR. For now we can merge with platform level flag but we need to make this a config model flag.

Reply:
@mushtaqak I've taken over this PR, and have a question about this setting.

As implemented, it will:
* show the `video_auto_advance` option on all video blocks
* but *only uses the video block setting* if [enabled both on the block and in the `FEATURE` flag](https://github.com/edx/edx-platform/pull/15803/files#diff-2386122d0b4952fd587ad8fa9556ef8dR335)

I would expect it to work more like [`show_reset_button`](https://github.com/open-craft/edx-platform/blob/363f31f967b93fe4a89ccfcfdcb29929b20365d2/common/lib/xmodule/xmodule/modulestore/inheritance.py#L202-L213) instead, and:
* show the `video_auto_advance` option on all video blocks
* use the `ENABLE_AUTOADVANCE_VIDEOS` feature flag to determine the default auto-advance setting for new video blocks when they're added to a course.

Reply:
As per my understanding here is how it should work:

If platform level flag (`ENABLE_AUTOADVANCE_VIDEOS`) is ON then show the `video_auto_advance` flag in advance settings. And if `video_auto_advance` is set in advance settings of a course then we should video_auto_advance option on all video blocks. 

Does this help ?

Reply:
Thanks @mushtaqak, that helps a lot.

I'm seeing one difference in my testing:

When `FEATURES['ENABLE_AUTOADVANCE_VIDEOS']` is False (which is the default), "Enable video auto-advance" shows up in Advanced Settings, with a default value of `False`.
Changing this Advanced Setting has no effect on the video blocks in the course.

But, when I set `FEATURES['ENABLE_AUTOADVANCE_VIDEOS'] = True`, restart the edxapp service, AND set the course Advanced Setting > "Enable video auto-advance" to `true`, then I see the "Auto-advance" button in the video unit for Studio and LMS.

So what I'm hearing is, we shouldn't show "Enable video auto-advance" in Advanced Settings unless `FEATURES['ENABLE_AUTOADVANCE_VIDEOS'] = True`?

Reply:
>So what I'm hearing is, we shouldn't show "Enable video auto-advance" in Advanced Settings unless FEATURES['ENABLE_AUTOADVANCE_VIDEOS'] = True?

Yes, you got that correct, if `FEATURES['ENABLE_AUTOADVANCE_VIDEOS']` is `False`, then `Enable video auto-advance` should not been shown. 

The only work of `FEATURES['ENABLE_AUTOADVANCE_VIDEOS']` is to let courses give `Enable video auto-advance` option in `Advance settings`. Rest is on `Enable video auto-advance` settings to show or not on video xblock.

Reply:
@mushtaqak I couldn't find any examples of course Advanced Settings which are only shown if a `FEATURE` (or other app `settings`) flag is set, and so I don't think this is possible?  I've only found cases where the default `value` of the Advanced Setting is determined by a `FEATURE` flag.

e.g., the ["Mobile Course Available" course Advanced Setting](https://github.com/edx/edx-platform/blob/b17cd9b992f13dacbcf939cc641a23e1842b9dbd/common/lib/xmodule/xmodule/course_module.py#L342-L347) can be set to default to `true` or `false` by using the app setting [`DEFAULT_MOBILE_AVAILABLE`](https://github.com/edx/edx-platform/blob/6db93fc791fd2fb52ce705d47320e7868a937587/cms/envs/aws.py#L113-L117).

Would it be ok to make `ENABLE_AUTOADVANCE_VIDEOS` work like this instead?

Reply:
I think it is working as I had mentioned.

The platform-wide flag is being initiallized though env.json. Also mentioned in step # 11 in testing instructions.
> In lms.env.json, add "AUTOADVANCE_VIDEOS": true to the FEATURES section.

If we set it false, we won't be shown the `AUTOADVANCE_VIDEOS` in advance settings page.
 I don't know if I have ssh access to sandbox.

Reply:
@pomegranited ^^

Reply:
@mushtaqak I've disabled the feature flag in `*.env.json` and restarted the LMS/CMS so you can see what happens when the flag is disabled.  I don't think it behaves how you want it to.

Reply:
@pomegranited [this](https://github.com/edx/edx-platform/blob/e23e3a0486be46b71f65ef56852446f21438fd04/cms/djangoapps/models/settings/course_metadata.py#L81-L81) is how we should've let this settings to not to show up on a Course's Advanced Settings.

Reply:
Ahhh...  thanks @Qubad786!  Sorry that didn't make it in this change.